server/sonar-server/src/main/java/org/sonar/server/test/index/TestIndexer.java

```diff
@@ -49,9 +50,23 @@ public TestIndexer(System2 system2, DbClient dbClient, EsClient esClient) {
     this.dbClient = dbClient;
   }
 
-  public void index(final String projectUuid) {
-    deleteByProject(projectUuid);
-    super.index(lastUpdatedAt -> doIndex(lastUpdatedAt, projectUuid));
+  @Override
+  public void indexProject(String projectUuid, Cause cause) {
+    switch (cause) {
+      case PROJECT_CREATION:
+        // no need to index, not tests at that time
+        break;
+      case PROJECT_KEY_UPDATE:
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This case's code block is the same as the block for the case on line 56. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1871)